src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -504,68 +506,63 @@ private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt)
-                LogFileGroup logFileGroup = attemptLogInformation.getLogFileGroup().syncDirectory();
-                Optional<LogFile> file1 = logFileGroup.getFile(fileHash);
-                // TODO: this is temporary for not changing the original logic, will be well handled.
-                if (!file1.isPresent()) {
-                    logger.atDebug().log("The fileHash does not exist in directory");
-                    return;
-                }
-                file = file1.get();
-                fileName = file.getAbsolutePath();
-                // TODO: the following logic is only for passing this small PR while not changing the current context.
-                //  We will grab the fileHash in the future.
-                Optional<LogFile> activeFile = logFileGroup.getActiveFile();
-                if (activeFile.isPresent()) {
-                    isActiveFile = activeFile.get().fileEquals(file);
-                }
-            } catch (InvalidLogGroupException e) {
-                logger.atDebug().cause(e).log();
+        try {
+            LogFileGroup logFileGroup = attemptLogInformation.getLogFileGroup().syncDirectory();
+            if (!logFileGroup.getFile(fileHash).isPresent()) {
+                logger.atTrace().log("The fileHash does not exist in directory");
                 return;
             }
-        }
-        // If we have completely read the file, then we need add it to the completed files list and remove it
-        // it (if necessary) for the current processing list.
-        String componentName = attemptLogInformation.getComponentName();
-        if (!isActiveFile && file.length() == cloudWatchAttemptLogFileInformation.getBytesRead()
-                + cloudWatchAttemptLogFileInformation.getStartPosition()) {
-            Set<String> completedFileNames = completedLogFilePerComponent.getOrDefault(componentName, new HashSet<>());
-            completedFileNames.add(fileName);
-            completedLogFilePerComponent.put(componentName, completedFileNames);
-            if (currentProcessingLogFilePerComponent.containsKey(componentName)) {
-                CurrentProcessingFileInformation fileInformation = currentProcessingLogFilePerComponent
-                        .get(componentName);
-                if (fileInformation.fileName.equals(fileName)) {
-                    currentProcessingLogFilePerComponent.remove(componentName);
+            LogFile file = logFileGroup.getFile(fileHash).get();
+            Optional<LogFile> activeFile = logFileGroup.getActiveFile();
+            boolean isActiveFile = false;
+            if (activeFile.isPresent()) {
+                isActiveFile = activeFile.get().fileEquals(file);
+            }
+            //This is deprecated value in versions greater than 2.2, but keep it here to avoid
+            // upgrade-downgrade issues.
+            String fileName = activeFile.get().getAbsolutePath();
```

Comment:
⚠️ CAREFUL with this one  - you are always making the fileName be the active file name. That is not how it used to work. This fileName was from the parameter that was being passed so this should basically be the `logFile.get().getAbsolutePath()`